partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/DynamicTyper.kt

```diff
@@ -0,0 +1,362 @@
+                FLOAT32 to FLOAT32,
+                FLOAT64 to FLOAT64,
+            )
+            graph[INT32] = edges(
+                INT8 to INT32,
+                INT16 to INT32,
+                INT32 to INT32,
+                INT64 to INT64,
+                INT to INT,
+                DECIMAL to DECIMAL,
+                DECIMAL_ARBITRARY to DECIMAL_ARBITRARY,
+                FLOAT32 to FLOAT32,
+                FLOAT64 to FLOAT64,
+            )
+            graph[INT64] = edges(
+                INT8 to INT64,
+                INT16 to INT64,
+                INT32 to INT64,
+                INT64 to INT64,
+                INT to INT,
+                DECIMAL to DECIMAL,
+                DECIMAL_ARBITRARY to DECIMAL_ARBITRARY,
+                FLOAT32 to FLOAT32,
+                FLOAT64 to FLOAT64,
+            )
+            graph[INT] = edges(
+                INT8 to INT,
+                INT16 to INT,
+                INT32 to INT,
+                INT64 to INT,
+                INT to INT,
+                DECIMAL to DECIMAL,
+                DECIMAL_ARBITRARY to DECIMAL_ARBITRARY,
+                FLOAT32 to FLOAT32,
+                FLOAT64 to FLOAT64,
+            )
+            graph[DECIMAL] = edges(
+                INT8 to DECIMAL,
+                INT16 to DECIMAL,
+                INT32 to DECIMAL,
+                INT64 to DECIMAL,
+                INT to DECIMAL,
+                DECIMAL to DECIMAL,
+                DECIMAL_ARBITRARY to DECIMAL_ARBITRARY,
+                FLOAT32 to FLOAT32,
+                FLOAT64 to FLOAT64,
+            )
+            graph[DECIMAL_ARBITRARY] = edges(
+                INT8 to DECIMAL_ARBITRARY,
+                INT16 to DECIMAL_ARBITRARY,
+                INT32 to DECIMAL_ARBITRARY,
+                INT64 to DECIMAL_ARBITRARY,
+                INT to DECIMAL_ARBITRARY,
+                DECIMAL to DECIMAL_ARBITRARY,
+                DECIMAL_ARBITRARY to DECIMAL_ARBITRARY,
+                FLOAT32 to FLOAT32,
+                FLOAT64 to FLOAT64,
+            )
+            graph[FLOAT32] = edges(
+                INT8 to FLOAT32,
+                INT16 to FLOAT32,
+                INT32 to FLOAT32,
+                INT64 to FLOAT32,
+                INT to FLOAT32,
+                DECIMAL to FLOAT32,
+                DECIMAL_ARBITRARY to FLOAT32,
+                FLOAT32 to FLOAT32,
+                FLOAT64 to FLOAT64,
+            )
+            graph[FLOAT64] = edges(
+                INT8 to FLOAT64,
+                INT16 to FLOAT64,
+                INT32 to FLOAT64,
+                INT64 to FLOAT64,
+                INT to FLOAT64,
+                DECIMAL to FLOAT64,
+                DECIMAL_ARBITRARY to FLOAT64,
+                FLOAT32 to FLOAT64,
+                FLOAT64 to FLOAT64,
+            )
+            graph[CHAR] = edges(
+                CHAR to CHAR,
+                STRING to STRING,
+                SYMBOL to STRING,
```

Comment:
More or less "string" as far as our ability to express types goes..

```
a) If any of the data types in DTS is character string, then all data types in DTS shall be
character string, and all of them shall have the same character repertoire.

...

i) If any of the data types in DTS is character large object string, then the result data
type is character large object string with maximum length in characters equal to the
maximum of the lengths in characters and maximum lengths in characters of the data
types in DTS.

ii) If any of the data types in DTS is variable-length character string, then the result data
type is variable-length character string with maximum length in characters equal to the
maximum of the lengths in characters and maximum lengths in characters of the data
types in DTS.

iii) Otherwise, the result data type is fixed-length character string with length in characters
equal to the maximum of the lengths in characters of the data types in DTS.
```

Based off of SQL it's pretty-much `CLOB > VARCHAR > CHAR` but this implementation doesn't include the length constraints.